src/examples/GLTFs.tsx

```diff
@@ -179,68 +189,255 @@ export const gltfRoutes = [
+    // use Three JS Loader
+    setComponent(modelEntity, NameComponent, 'Three-Loader')
+    setComponent(modelEntity, ModelComponent, {
+      cameraOcclusion: true,
+      src: src
+    })
+    setComponent(modelEntity, ShadowComponent, { receive: false })
+    setVisibleComponent(modelEntity, true)
+    const modelTransform = getComponent(modelEntity, TransformComponent)
+    modelTransform.position.set(offsetVec.x, offsetVec.y, offsetVec.z)
+    if (scale) modelTransform.scale.set(scale, scale, scale)
+  }, [src])
 
-  useEffect(() => {
-    if (!assetEntity || !props.scale) return
+  const playAnimation = (component: State<ComponentType<typeof AnimationComponent>> | undefined) => {
+    const animationComponent = component?.value
+    if (!animationComponent?.animations || !animationClip) return
 
-    setComponent(assetEntity, TransformComponent, { scale: new Vector3().setScalar(props.scale) })
-  }, [assetEntity, props.scale])
+    const clips = animationComponent.animations as AnimationClip[]
+    const clip = AnimationClip.findByName(clips, animationClip)
+    if (!clip) return console.warn('Clip not found:', animationClip)
+
+    const action = animationComponent.mixer.clipAction(clip)
+    action.play()
+  }
 
   useEffect(() => {
-    if (!animationComponent?.value?.animations || !props.animationClip) return
+    playAnimation(gltfAnimation)
+  }, [gltfAnimation, animationClip])
 
-    const clips = animationComponent?.value?.animations as AnimationClip[]
+  useEffect(() => {
+    playAnimation(modelAnimation)
+  }, [modelAnimation, animationClip])
 
-    const clip = AnimationClip.findByName(clips, props.animationClip)
-    if (!clip) return console.warn('Clip not found:', props.animationClip)
+  return null
+}
 
-    console.log(clip)
+export default function GLTFViewer(props: {
+  src: string
+  scale?: number
+  offset?: number | { x?: number; y?: number; z?: number }
+  light?: boolean
+  animationClip?: string
+}) {
+  const sceneEntity = useRouteScene()
 
-    const action = animationComponent.value.mixer.clipAction(clip)
-    action.play()
-  }, [animationComponent?.value?.animations, props.animationClip])
+  useEffect(() => {
+    const bgColor = document.body.style.backgroundColor
+    document.body.style.backgroundColor = 'gray'
+    getMutableState(RendererState).gridVisibility.set(true)
+    getMutableState(RendererState).physicsDebug.set(true)
+
+    return () => {
+      document.body.style.backgroundColor = bgColor
+    }
+  }, [])
 
   useEffect(() => {
-    if (!props.light) return
+    if (!props.light || !sceneEntity) return
 
     const entity = createEntity()
     setComponent(entity, UUIDComponent, 'directional light' as EntityUUID)
     setComponent(entity, NameComponent, 'Directional Light')
     setComponent(entity, TransformComponent, { rotation: new Quaternion().setFromEuler(new Euler(2, 5, 3)) })
-    setComponent(entity, EntityTreeComponent, { parentEntity: getState(EngineState).originEntity })
```

Comment:
i think the ambient light should not be attached to the scene entity, since it is not a part of the scene. attaching to the origin entity will still render it, but it wont interfere with anything scene specific